### ISSUE TITLE
New EA Games?

### ISSUE THREAD

**Issue by d10sfan** [luxtorpeda-dev/packages#1139]
1. https://github.com/OpenSAGE/OpenSAGE
2. https://www.openra.net/download/ (detect steam assets?)
3. ~https://github.com/TheAssemblyArmada/Vanilla-Conquer for original games (look at existing vanilla-conquer use)~
4. alpha centurai?

**Comment by rea987** [comment]
1. https://github.com/OpenRA/ra2 (Red Alert 2)
2. https://github.com/MustaphaTR/Romanovs-Vengeance (modded Red Alert 2 mp)
3. https://github.com/ABrandau/Shattered-Paradise-SDK (modded Shattered Paradise mp)

**Comment by d10sfan** [comment]
Once https://github.com/OpenRA/OpenRA/pull/21355 is ready, then I could add that for the original games. Working on adding Vanilla Conquer

For RA2 & OpenRA, it seems like that hasn't been updated in a while. Will investigate the other two in the future.

**Comment by d10sfan** [comment]
I've added vanilla-conquer to both of the new original steam versions, so that should be working now

**Comment by rea987** [comment]
@d10sfan

> For RA2 & OpenRA, it seems like that hasn't been updated in a while. Will investigate the other two in the future.

@ondrejmirtes' RA2 fork seems to be few commits ahead of the main branch and offers an AppImage build.

https://github.com/ondrejmirtes/ra2

**Comment by ondrejmirtes** [comment]
@rea987 Yeah, I made the build work in GitHub Actions. But I wholly recommend using https://github.com/MustaphaTR/Romanovs-Vengeance which is actively developed and really awesome! Just spent the last weekend playing it with friends at a LAN party :)

**Comment by hazelnot** [comment]
> I've added vanilla-conquer to both of the new original steam versions, so that should be working now

The games start, but the folder structure is unsuitable for VanillaRA. Red Alert, for example, crashes when trying to start a campaign, and is unable to read any of the expansion pack campaigns or multiplayer map files.

I would also advise against using OpenRA as anything other than an optional secondary choice, as OpenRA is more of a fan remake that reuses the assets and basic gameplay, but the mechanics and balance are very different in practice to the original games.

**Comment by d10sfan** [comment]
Interesting, I'll look into that soon

**Comment by rea987** [comment]
@hazelnot

> I would also advise against using OpenRA as anything other than an optional secondary choice, as OpenRA is more of a fan remake that reuses the assets and basic gameplay, but the mechanics and balance are very different in practice to the original games.

Many if not the most games here use engine re-implementations which aren't exactly the same with the original game. I don't see an issue with OpenRA as an addition. Considering that none of the 2D C&C titles' online multiplayer work properly without patches and/or 3rd party clients, OpenRA is a great convenience for the players.

**Comment by hifi** [comment]
> Considering that none of the 2D C&C titles' online multiplayer work properly without patches and/or 3rd party clients, OpenRA is a great convenience for the players.

We should have LAN working for both TD and RA in Vanilla Conquer but it doesn't have the convenience of easy online multiplayer.

It all boils down to the preferred order and accurate description of both of the engines. VC is based on the original source and strictly about original gameplay while making them portable. ORA is about other conveniences and re-imagined gameplay and is much more mature as a project with a lot more developers behind it.

**Comment by hazelnot** [comment]
It's really a matter of prioritizing singleplayer or multiplayer. OpenRA is very multiplayer-focused, and has a very hardcore playerbase in my experience, while missing or reimagining big chunks of the campaigns.

I've never really played C&C online because pretty much everyone doing that has been playing for 20 years, so OpenRA despite the quality of life improvements gets in my opinion even less beginner-friendly.

As far as I'm aware most people who play C&C these days tend to stick with singleplayer for this reason, which is why I said that OpenRA shouldn't be used as a primary/default option.

**Comment by hazelnot** [comment]
I looked into RA1 a bit more and talked to some of the vanilla conquer devs

The game was repackaged, and currently vanilla conquer itself doesn't support it out of the box due to some of the different filenames.

I managed to fix the filestructure of the `linuxdata-ra` folder by creating some new folders and redoing the symlinks (as currently luxtorpeda assumes it's working with the Remastered Collection version of the game)

Basically, `main1.mix` corresponds to `allied/main.mix`, `main2.mix` to `soviet/main.mix`, `main3.mix` to `counterstrike/main.mix` and `main4.mix` to `aftermath/main.mix`. The rest of the files are in the game's root directory, so for example `../Data/CNCDATA/RED_ALERT/CD1/EXPAND.MIX` simply needs to be redirected to `../EXPAND.MIX`

At least until VC gets updated to support this packaging scheme.

**Comment by d10sfan** [comment]
@hazelnot nice! Can you send a screenshot or copy paste the directory listing for the games so I can verify how it looks? I can update the setup to match for the non remastered games

**Comment by hazelnot** [comment]
Looks like this. `HIRES.MIX` and `LORES.MIX` aren't necessary as they're already packed into a different MIX, though I'm not really sure about HIRES1.MIX and LORES1.MIX, which are apparently the icons for units added by The Aftermath. They may or may not be necessary, though I'm not sure.

![image](https://github.com/luxtorpeda-dev/packages/assets/10992973/69997624-2356-4d35-9cdf-397afda1736f)

**Comment by d10sfan** [comment]
@hazelnot I've pushed up some changes for the original RA, if you can test it out when you have a chance. I didn't see either of those pal files, but it seems to run ok without them.

**Comment by d10sfan** [comment]
I've also added support to the original C&C for OpenRA, and it'll discover the steam installation and put it in the right place, other than the movies.

The steps to get it for manual for ra is pretty complicated (https://gist.github.com/abcdefg30/a4fe1110c2727c3960e340ae2ce45bc0), so I'll leave that alone until the auto-detect is ready (and re-visit cnc)

**Comment by d10sfan** [comment]
RA2 (https://github.com/MustaphaTR/Romanovs-Vengeance?tab=readme-ov-file) can be added by creating a link between the ra2 folder and ~/.config/openra/Content/ra2, then grabbing the appimage from https://github.com/MustaphaTR/Romanovs-Vengeance?tab=readme-ov-file, and following the same as the cnc openra.

Looks like tiberium sun could be done similarly with: https://github.com/ABrandau/Shattered-Paradise-SDK/releases

**Comment by d10sfan** [comment]
RA2 and Tiberium Sun have been added, ra2 uses the steam assets, tiberium sun you can use the quick install for now.

I believe that's all the games covered that make sense, as OpenSAGE is still a bit too early, but if there's anything else, feel free to comment here or add a new issue. Thanks!

**Comment by rea987** [comment]
@d10sfan Having Romanov's Vengeance for RA2 is great as it offer new balancing options and units but it's not exactly RA2 or YR but rather an different multiplayer take on the game. It would be nice to have modern and more authentic multiplayer remakes of RA2 and YR. Hence I believe following OpenRA mods can be added;

- RA2: https://github.com/OpenRA/ra2

It is called as Red Alert 2 ({DEV_VERSION}) on MP servers list. I compiled this one with Mono 6.12.0. It works fine with Steam assets and there's even an online server available for this build.

- https://github.com/cookgreen/Yuris-Revenge/releases

This mod offers Yuri faction and AppImage builds are available.

**Comment by rea987** [comment]
@d10sfan Yes, listed .mix files need to be put in `~/.config/openra/Content/ra2` and `~/.config/openra/Content/yr` .

https://github.com/cookgreen/Yuris-Revenge/wiki/Install-the-content

**Comment by d10sfan** [comment]
Thanks that looks like it did the trick. I added yr to the supported ones, for [RA2](https://github.com/OpenRA/ra2) I'll need to wait for them to release an appimage, as packaging all of the necessary libraries for that project would not be feasible at this point.
